sklearn/datasets/tests/test_openml.py

```diff
@@ -797,6 +797,9 @@ def test_decode_iris(monkeypatch):
     _test_features_list(data_id)
 
 
+@pytest.mark.filterwarnings(
+    "ignore:Multiple active versions of the dataset matching the name"
+)
```

Comment:
I think we should explain why it's fine to ignore this warning with a comment such as:

```suggestion
@pytest.mark.filterwarnings(
    # This test is intentionally passing both data_id and data_name + version in
    # which case and this causes _fetch_dataset_from_openml to trigger a spurious
    # warning.
    "ignore:Multiple active versions of the dataset matching the name"
)
```

@glemaitre I am not sure about this explanation. You investigating the cause.

Isn't there a simple way to just avoid triggering the warning in the first place? If no please expand on the suggestion explanation because I do not find it satisfying.

Is the problem related to monkeypatching?

Reply:
Also do we really want to pass both data_id and data_name + versions in those tests? Why not just data_id and put the name as a comment?

Reply:
Actually discussion with @glemaitre I think we should instead change `_fetch_dataset_from_openml` to avoid raising that specific warning in the first place:
```python
with with warnings.catch_warnings():
    warnings.filterwarnings("ignore", category=UserWarning)
    fetch_openml(...)
```

Then we will also need to update `test_fetch_openml_iris` to directly call `fetch_openml("iris")`  instead of using  `_fetch_dataset_from_openml`.

Reply:
> Assuming the tests pass, this looks much better to me :)

Two more warnings remain, though, regarding the Australian dataset's version 1 being outdated.